.github/workflows/compatibility.yml

```
@@ -0,0 +1,66 @@
+name: Compatibility Tests
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  schedule:
+    # Run weekly on Sundays at 3 AM UTC
+    - cron: "0 3 * * 0"
+
+jobs:
+  go-compatibility:
+    name: Go ${{ matrix.go-version }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      matrix:
+        go-version: ["1.21", "1.22", "1.23", "1.24"]
```

I'm a bit worried these ones would be deprecated at some point,this list would need to be maintained.

So I would like to suggest you adding at least the stable and oldstable ones.


```suggestion
        go-version: ["1.21", "1.22", "1.23", "1.24", "oldstable", "stable"]
```

---

@ccoVeille Got it — you mean something like this?

```
go-version: ["1.21", "oldstable", "stable"]
```

---

Currently:

- go.mod of project refers to Go 1.23
- stable point to Go 1.24
- oldstable point to Go 1.23
- Go 1.25 is planned for August, and it will become the stable, and 1.24 the old stable

I would use "go1.23", "oldstable", "stable"

---

If the go.mod memtion go 1.23 it means it cannot be used as a package with go 1.21 or 1.22.

I'm not even sure the using go test on Go 1.21 won't make it run on Go 1.23.

But this goes beyond the scope of your PR

For me, the problem is the README.

But you can run test on Go 1.21 1.22

Either tests are useless as they would use Go 1.23, either they will be used if go.mod is updated to downgrade to Go 1.21